### DISCUSSION TITLE
Adding comments when editor is read-only

### DISCUSSION THREAD

**Discussion by varun-r-boop** [sereneinserenade/tiptap-comment-extension#15] [Q&A]
Do we have the possibility to add comments even when the editor is not editable, currently I have my comment inside my bubble menu, but I have trouble showing the bubble menu in read-only because the `shouldShow` is not being triggered when the editor is not editable.Is there any way we could achieve this @sereneinserenade 
Thanks :)

**Comment by sereneinserenade** [comment]
Either you need to provide your own [`shouldShow`](https://tiptap.dev/docs/editor/extensions/functionality/bubble-menu#shouldshow) property to BubbleMenu. If that doesn't work, you'd need to figure out which code in [bubble-menu-plugin](https://github.com/ueberdosis/tiptap/blob/develop/packages/extension-bubble-menu/src/bubble-menu-plugin.ts) is stopping from `shouldShow` being called when editor is read-only & modify it to meet your needs.

**Reply by varun-r-boop** [reply]
I looked into the code which was blocking the bubble it was obviously the `isEditable` of the editor , but even though if I am able to access the bubble I still wouldn't be able to add the comments if the editor is not editable since I am using the `Marks` approach 
Currently the approach Im trying is to make the editor read-only without setting the  `isEditable` to false by blocking all the actions except the selection @sereneinserenade
